src/main/java/me/i509/fabric/bulkyshulkies/extension/QuickShulkerHook.java

```diff
@@ -0,0 +1,116 @@
+		ContainerProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_11x7_CONTAINER, QuickShulkerHook::create11x7);
+		ContainerProviderRegistry.INSTANCE.registerFactory(QS$SHULKER_13x7_CONTAINER, QuickShulkerHook::create13x7);
+
+		QuickOpenableRegistry.register(((player, stack) -> ContainerProviderRegistry.INSTANCE.openContainer(QS$SHULKER_SCROLLABLE_CONTAINER, player,
+				writer -> writer.writeInt(player.inventory.getSlotWithStack(stack)))),
+				CopperShulkerBoxBlock.class, IronShulkerBoxBlock.class, SilverShulkerBoxBlock.class, CursedSlabShulkerBox.class);
+
+		QuickOpenableRegistry.register(((player, stack) -> ContainerProviderRegistry.INSTANCE.openContainer(QS$SHULKER_13x7_CONTAINER, player,
+				writer -> writer.writeInt(player.inventory.getSlotWithStack(stack)))),
+				ObsidianShulkerBoxBlock.class, PlatinumShulkerBoxBlock.class);
+
+		QuickOpenableRegistry.register(((player, stack) -> ContainerProviderRegistry.INSTANCE.openContainer(QS$SHULKER_11x7_CONTAINER, player,
+				writer -> writer.writeInt(player.inventory.getSlotWithStack(stack)))),
+				DiamondShulkerBoxBlock.class);
+
+		QuickOpenableRegistry.register(((player, stack) -> ContainerProviderRegistry.INSTANCE.openContainer(QS$SHULKER_9x7_CONTAINER, player,
+				writer -> writer.writeInt(player.inventory.getSlotWithStack(stack)))),
+				GoldShulkerBoxBlock.class);
+	}
+
+	public static GenericContainer13x7 create13x7(int syncId, Identifier identifier, PlayerEntity player, PacketByteBuf buf) {
+		int invSlot = buf.readInt();
+		ItemStack stack = player.inventory.getInvStack(invSlot);
+		return new GenericContainer13x7(syncId, ShulkerBoxSlot::new, player.inventory, new ItemStackInventory(stack, getSlotCount(stack)), new TranslatableText("container.shulkerBox"));
+	}
+
+	public static GenericContainer11x7 create11x7(int syncId, Identifier identifier, PlayerEntity player, PacketByteBuf buf) {
+		int invSlot = buf.readInt();
+		ItemStack stack = player.inventory.getInvStack(invSlot);
+		return new GenericContainer11x7(syncId, ShulkerBoxSlot::new, player.inventory, new ItemStackInventory(stack, getSlotCount(stack)), new TranslatableText("container.shulkerBox"));
+	}
+
+	public static GenericContainer9x7 create9x7(int syncId, Identifier identifier, PlayerEntity player, PacketByteBuf buf) {
+		int invSlot = buf.readInt();
+		ItemStack stack = player.inventory.getInvStack(invSlot);
+		return new GenericContainer9x7(syncId, ShulkerBoxSlot::new, player.inventory, new ItemStackInventory(stack, getSlotCount(stack)), new TranslatableText("container.shulkerBox"));
+	}
+
+	public static ScrollableContainer createScrollable(int syncId, Identifier identifier, PlayerEntity player, PacketByteBuf buf) {
+		int invSlot = buf.readInt();
+		ItemStack stack = player.inventory.getInvStack(invSlot);
+		return new ScrollableContainer(syncId, ShulkerBoxSlot::new, player.inventory, new ItemStackInventory(stack, getSlotCount(stack)), new TranslatableText("container.shulkerBox"));
```

Comment:
Usually we would want the text in the container to match the name of the block entity. If you could make a ShulkerTexts class in the registry package and refer to the text constants there it would work fine.